tests/kafkatest/tests/streams/streams_smoke_test.py

```diff
@@ -31,48 +31,80 @@ def __init__(self, test_context):
-        Start a few smoke test clients, then repeat start a new one, stop (cleanly) running one a few times.
-        Ensure that all results (stats on values computed by Kafka Streams) are correct.
-        """
+    @cluster(num_nodes=8)
+    @matrix(eos=[True, False], crash=[True, False])
+    def test_streams(self, eos, crash):
+        #
+        if eos:
+            processor1 = StreamsSmokeTestEOSJobRunnerService(self.test_context, self.kafka)
+            processor2 = StreamsSmokeTestEOSJobRunnerService(self.test_context, self.kafka)
+            processor3 = StreamsSmokeTestEOSJobRunnerService(self.test_context, self.kafka)
+        else:
+            processor1 = StreamsSmokeTestJobRunnerService(self.test_context, self.kafka)
+            processor2 = StreamsSmokeTestJobRunnerService(self.test_context, self.kafka)
+            processor3 = StreamsSmokeTestJobRunnerService(self.test_context, self.kafka)
 
-        self.driver.start()
 
-        self.processor1.start()
-        self.processor2.start()
 
-        time.sleep(15)
+        with processor1.node.account.monitor_log(processor1.STDOUT_FILE) as monitor1:
+            processor1.start()
+            monitor1.wait_until('REBALANCING -> RUNNING',
+                               timeout_sec=60,
+                               err_msg="Never saw 'REBALANCING -> RUNNING' message " + str(processor1.node.account)
+                               )
 
-        self.processor3.start()
-        self.processor1.stop()
+            self.driver.start()
 
-        time.sleep(15)
+            monitor1.wait_until('processed',
+                                timeout_sec=30,
+                                err_msg="Didn't see any processing messages " + str(processor1.node.account)
+                                )
 
-        self.processor4.start()
+            # make sure we're not already done processing (which would invalidate the test)
+            self.driver.node.account.ssh("! grep 'Result Verification' %s" % self.driver.STDOUT_FILE, allow_fail=False)
+
+            processor1.stop_nodes(not crash)
+
+        with processor2.node.account.monitor_log(processor2.STDOUT_FILE) as monitor3:
+            processor2.start()
+            monitor3.wait_until('REBALANCING -> RUNNING',
+                                timeout_sec=120,
+                                err_msg="Never saw 'REBALANCING -> RUNNING' message " + str(processor2.node.account)
+                                )
+            monitor3.wait_until('processed',
+                                timeout_sec=30,
+                                err_msg="Didn't see any processing messages " + str(processor2.node.account)
+                                )
+
+        # make sure we're not already done processing (which would invalidate the test)
+        self.driver.node.account.ssh("! grep 'Result Verification' %s" % self.driver.STDOUT_FILE, allow_fail=False)
+
+        processor2.stop_nodes(not crash)
+
+        processor3.start()
```

Comment:
Don't we want to at least `processor3` starts successfully here?

Reply:
We could wait on that, it might make the test more readable. But it's not strictly necessary, since if it doesn't start, then we won't finish processing.

I'll add it for readability.

Reply:
Done, it occurred to me during the change that it also has the side benefit of making sure that the third processor does process some events.